src/robot/client.ts

```diff
@@ -386,18 +392,23 @@ export class RobotClient extends EventDispatcher implements Robot {
       this.peerConn = undefined;
     }
     this.sessionManager.reset();
+    this.emit(MachineConnectionEvent.DISCONNECTED, {});
   }
 
   public isConnected(): boolean {
     return this.peerConn?.iceConnectionState === 'connected';
   }
 
+  // TODO: refactor due to cognitive complexity
```

Comment:
(minor) if this is something that will be revisited in a subsequent PR, can we create a ticket and link to it here (same with below `TODO` comment).

Reply:
Will do!